src/plugins/UserStats.js

```diff
@@ -14,41 +15,45 @@ export default class UserStats extends Plugin {
     }
 
     proxy(eventName, message) {
-        if (message.from.username) {
-            const chatId = message.chat.id;
-            const userId = message.from.id;
-            if (!this.db["stat" + chatId]) {
-                this.db["stat" + chatId] = {};
-                this.db["stat" + chatId].totalMessageCount = 0;
-            }
-            if (!this.db["stat" + chatId][userId]) {
-                this.db["stat" + chatId][userId] = {
-                    id: userId,
-                    username: message.from.username,
-                    messageCount: 0
-                };
-            }
-            this.db["stat" + chatId][userId].messageCount++;
-            this.db["stat" + chatId].totalMessageCount++;
+        if (!message.from.username) return;
+        const chatId = message.chat.id;
+        const userId = message.from.id;
+        if (!this.db["stat" + chatId]) {
+            this.db["stat" + chatId] = {};
+            this.db["stat" + chatId].totalMessageCount = 0;
+        }
+        if (!this.db["stat" + chatId][userId]) {
+            this.db["stat" + chatId][userId] = {
+                userId,
+                username: message.from.username,
+                messageCount: 0
+            };
         }
+        this.db["stat" + chatId][userId].messageCount++;
+        this.db["stat" + chatId].totalMessageCount++;
         return Promise.resolve(message);
     }
 
     onText(message, reply) {
-        if (message.text === '/UserStats') {
-            let text = `Total Messages:\n\n`;
-            const statsObject = this.db["stat" + message.chat.id];
-            const totalCount = statsObject.totalMessageCount;
-            for (let key in statsObject) {
-                if (key !== 'totalMessageCount') {
-                    const username = statsObject[key].username;
-                    const id = statsObject[key].id;
-                    const count = statsObject[key].messageCount;
-                    const percentage = (count / totalCount * 100).toFixed(4);
-                    text += `${username} [${id}]: ${count} (${percentage}%)\n`;
-                }
+        const parts = Util.parseCommand(message.text, "userstats");
+        if (!parts) return;
+        let text = `Total Messages:\n\n`;
+        const statsObject = this.db["stat" + message.chat.id];
+        const totalCount = statsObject.totalMessageCount;
+        const userList = Object.keys(statsObject)
+            .map(item => {
+                return statsObject[item];
+            }).filter(item => {
+                return typeof item === "object";
+            }).sort((a, b) => {
+                return b.messageCount - a.messageCount;
+            });
+        for (let user of userList) {
+            if (user) {
+                const percentage = (user.messageCount / totalCount * 100).toFixed(4);
+                text += `${user.username} [${user.userId}]: ${user.messageCount} (${percentage}%)\n`;
```

Comment:
You most likely have to delete plugin_UserStats in redis db since it is using object from before. I applied the syntactic sugar you suggested in different part of the code to https://github.com/avipatel91/Telegram-Bot-Node/blob/stats/src/plugins/UserStats.js#L27. id has changed to userId. Is that alright?